File: projects/collatz/makefile.py
run:
	RunCollatz.py < RunCollatz.in

diff:
	RunCollatz.py < RunCollatz.in > RunCollatz.tmp
	diff RunCollatz.out RunCollatz.tmp
	rm RunCollatz.tmp

test:
	TestCollatz.py

zip:
	zip -r Collatz.zip EADME.txt makefile Collatz.html Collatz.log Collatz.py \
	RunCollatz.in RunCollatz.out RunCollatz.py SphereCollatz.py \
	TestCollatz.py TestCollatz.out

clean:
	rm -f *.pyc
	rm -f *.tmp

File: projects/collatz/makefile.java
RunCollatz.class: RunCollatz.java Collatz.java
	javac -Xlint RunCollatz.java

run: RunCollatz.class
	java -ea RunCollatz < RunCollatz.in

diff: RunCollatz.class
	java -ea RunCollatz < RunCollatz.in > RunCollatz.tmp
	diff RunCollatz.out RunCollatz.tmp
	rm RunCollatz.tmp

TestCollatz.class: TestCollatz.java Collatz.java
	javac -Xlint TestCollatz.java

test: TestCollatz.class
	java -ea TestCollatz

clean:
	rm -f RunCollatz.class
	rm -f RunCollatz.tmp
    rm -f TestCollatz.class
